### DISCUSSION TITLE
connecting carta-python to client for scripting?

### DISCUSSION THREAD

**Discussion by aakepley** [idia-astro/carta-python#30] [Q&A]
I'm attempting to play around a bit with the python 3 scripting interface for CARTA.

I have an Apple M1 laptop running Big Sur. I've downloaded CARTA 3.0.0-beta.2b and installed is as directed including adding an alias to CARTA to my .zshrc file. I've installed  carta-python via https://carta-python.readthedocs.io/en/latest/ . I needed to also do a pip install  google-api-python-client  to be able to import the package.

I've started the carta viewer up from the command line including the "--grpc_port" command line parameter, but a GRPC SECURITY TOKEN isn't printed to the command line. Some information flashes very briefly when CARTA starts up, but so briefly I can't make out what it is.

Is there another way to get the GRPC SECURITY TOKEN?

**Comment by confluence** [comment]
Could you please paste the exact command you're using to start CARTA, as well as the resulting terminal output?

**Reply by aakepley** [reply]
Here it is:

/Applications/CARTA-v3.0.0-beta.2b.app/Contents/MacOS/CARTA-v3.0.0-beta.2b --grpc_port

There is no terminal output.

**Comment by confluence** [comment]
The `--grpc_port` option requires an argument (the number of the port). You can check the usage of all the options with the `--help` flag. The suggested port value is `50051`.

(I'm not sure why there is no terminal output -- you should be seeing an error message about the missing argument. That's what I got when I reproduced this  -- but I'm using the Ubuntu package; I'm not sure if the Mac executable suppresses the log output somehow. What exactly do you see in your terminal? Does the executable exit immediately without printing any output? Does it hang?)
